### PR TITLE
linux/wayland: Add support for pasting images

### PR DESCRIPTION
**Pull request by maan2003**
Release Notes:

-  You can now paste images into the Assistant Panel to include them as context on Linux wayland

### CODE REVIEW

**Comment by maan2003** [comment]
:smile: I should really try to compile before pushing stuff

**Comment by maxdeviant** [comment]
@maan2003 Bit of a random/unrelated question, I see you have these (seemingly) randomly-generated branch names:

Screenshot 2024-09-10 at 7 10 12 PM

I'm curious if you're using something like [`jj`](https://github.com/martinvonz/jj)?

**Comment by maan2003** [comment]
yes I am using jj. It is awesome. not having to think about branch names is nice

**Comment by maxdeviant** [comment]
> yes I am using jj. It is awesome. not having to think about branch names is nice

Cool! I've been meaning to check it out myself 😄

File: crates/gpui/src/platform/linux/wayland/clipboard.rs [hunk 1]
```diff
@@ -33,14 +37,30 @@ pub(crate) struct Clipboard {
     current_primary_offer: Option<DataOffer<ZwpPrimarySelectionOfferV1>>,
 }
 
+pub trait ReceiveData {
```
Comment (maan2003) [inline crates/gpui/src/platform/linux/wayland/clipboard.rs:40]: should be pub(crate)

**Comment by maan2003** [comment]
tested on sway - works

**Comment by mikayla-maki** [comment]
Thank you!

**Pull request merged**